tardis/conftest.py

```diff
@@ -56,6 +56,11 @@ def pytest_addoption(parser):
 
 
 def pytest_configure(config):
+    # A common tempdir for storing plots / PDFs and other slow test related data
+    # generated during execution.
+    tempdir_session = tempfile.mkdtemp()
+    config.option.tempdir = tempdir_session
+
     html_file = tempfile.NamedTemporaryFile(delete=False)
     # Html test report will be generated at this filepath by pytest-html plugin
     config.option.htmlpath = html_file.name
```

Comment:
So pytest-html requires `config.option.htmlpath` to be set to generate a report?
In any case, I'd store the html file handle in the config, too. Then you don't have to worry about deleting the temp file because it should go when `config` goes out of scope.